tools/update_flutter_engine

```diff
@@ -14,4 +14,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 base_dir="$(dirname "$0")"
-exec "$base_dir/../../flutter/bin/cache/dart-sdk/bin/dart" "$base_dir/dart_tools/bin/update_flutter_engine.dart" $@
+flutter_bin_dir="$base_dir/../../flutter/bin"
+dart_bin_dir="$flutter_bin_dir/cache/dart-sdk/bin"
+# Ensure that the Dart SDK has been downloaded.
+if [[ ! -e $dart_bin_dir ]]; then
```

Comment:
wrap variable in quotes

Reply:
Isn't one of the advantages of `[[ ]]` not needing to quote variables?

"Word splitting and filename expansion are not performed on the words between the [[ and ]]"
https://www.gnu.org/software/bash/manual/bashref.html#Conditional-Constructs

Reply:
Oh, woops!  I read that as single bracket.  Carry on